src/picongpu/include/particles/synchrotronPhotons/PhotonCreator.hpp

```diff
@@ -0,0 +1,319 @@
+                (this->curF_1[z] + float_X(1.5) * delta * chi * z * this->curF_2[z]);
+        }
+        else
+        {
+            // classical
+            const float_X z = float_X(2.0/3.0) * delta / chi;
+
+            return factor / delta * this->curF_1[z];
+        }
+    }
+
+    /** Get the *scaled* photon emission probability
+     *
+     * The scaling avoids an infrared divergence.
+     *
+     * @param z scaled and normalized (to the electron energy) photon energy
+     * @param chi quantum-nonlinearity parameter
+     * @param gamma electron gamma
+     */
+    DINLINE float_X emission_prob_scaled(
+        const float_X z,
+        const float_X chi,
+        const float_X gamma) const
+    {
+        return float_X(3.0) * z*z * emission_prob(z*z*z, chi, gamma);
+    }
+
+    /** Return the number of target particles to create from each source particle.
+     *
+     * Called for each frame of the source species.
+     *
+     * @param sourceFrame Frame of the source species
+     * @param localIdx Index of the source particle within frame
+     * @return number of particle to be created from each source particle
+     */
+    DINLINE unsigned int numNewParticles(FrameType& sourceFrame, int localIdx)
+    {
+        using namespace PMacc::algorithms;
+
+        PMACC_AUTO(particle, sourceFrame[localIdx]);
+
+        /* particle position, used for field-to-particle interpolation */
+        floatD_X pos = particle[position_];
+        const int particleCellIdx = particle[localCellIdx_];
+        /* multi-dim coordinate of the local cell inside the super cell */
+        DataSpace<TVec::dim> localCell(DataSpaceOperations<TVec::dim>::template map<TVec > (particleCellIdx));
+        /* interpolation of E-field on the particle position */
+        const fieldSolver::numericalCellType::traits::FieldPosition<FieldE> fieldPosE;
+        ValueType_E fieldE = Field2ParticleInterpolation()
+            (cachedE.shift(localCell).toCursor(), pos, fieldPosE());
+        /* interpolation of B-field on the particle position */
+        const fieldSolver::numericalCellType::traits::FieldPosition<FieldB> fieldPosB;
+        ValueType_B fieldB = Field2ParticleInterpolation()
+            (cachedB.shift(localCell).toCursor(), pos, fieldPosB());
+
+        const float3_X mom = particle[momentum_] / particle[weighting_];
+        const float_X mom2 = math::dot(mom, mom);
+        const float3_X mom_norm = mom * math::rsqrt(mom2);
+        const float_X mass = frame::getMass<FrameType>();
+
+        const float_X gamma = Gamma<>()(mom, mass);
+        const float3_X vel = mom / (gamma * mass); // low accuracy?
+
+        const float3_X lorentz = fieldE + math::cross(vel, fieldB);
+        const float_X lorentz2 = math::dot(lorentz, lorentz);
+        const float_X fieldE_long = math::dot(mom_norm, fieldE);
+
+        // effective magnetic strength
+        const float_X H_eff = math::sqrt(lorentz2 - fieldE_long*fieldE_long);
```

Comment:
Since you are in SI units, could this be an `B_eff`? Otherwise there is a conversion factor of `H = 1/mu_0 B` in SI units.